### PR TITLE
Domain Expansion: Coffin of the Ashen Mountain (retracting your mining MOD is bad, actually)

### PR DESCRIPTION
**Pull request by Hatterhat**
## About The Pull Request
If your MODsuit has an ash accretion module installed, retracting your MOD will cause you to lose its boons (storm resistance/armor/speedboost).

## How This Contributes To The Nova Sector Roleplay Experience
Encourages MOD usage. Reminds miners that they are mortal.

## Proof of Testing
image
## Changelog

:cl:
fix: Retracting your MODsuit with an ash accretion module enabled disables the benefits from the module.
/:cl:

### CODE REVIEW

**Comment by Zergspower** [comment]
This really should be an upstream change, and i'm not sure why it needs to be changed either given the deployed suit is so slow and speed is needed for survival

**Comment by Hatterhat** [comment]
>Beyond that, this isnt a fix

I'm pretty sure some previous PRs were made to attempt to address this as a bug and none of them ever fixed it, pre-schism. Something, something, the PRs that disabled using MOD modules that weren't flagged as allow-while-disabled. I'm pretty sure the ash module was never meant to be usable while disabled, so - now it is.

**Comment by Zergspower** [comment]
Retracted with Ash on
image

Applied with Ash on
image

Both lose to the basic bitch armor miners start with, armor isnt really the reason anyone buys this modsuit as it's strictly to get the immunity from the storm with the bonus of speed - else you're losing out on armor if you deploy it

**Comment by Hatterhat** [comment]
> Both lose to the basic bitch armor miners start with, armor isnt really the reason anyone buys this modsuit as it's strictly to get the immunity from the storm with the bonus of speed - else you're losing out on armor if you deploy it

The issue _is_ that the stormproofing and speed-boost are, effectively, free, because you can retract the MOD (a Nova change), and the additional utility from having a mining MOD on your back provides some additional benefits depending on what you have installed.

You wear the explorer suit or other Lavaland gear for pure armor.
You wear the MOD for the utility (sphere form, crusher-on-suit, etc. etc.).

**Comment by Ephemeralis** [comment]
It's pretty obvious that the bonus isn't intended to be given while disabled.

**Comment by Namwty** [comment]
F to some miners who relied on this for bubblegum, due to natural movemen speed  slowdown compared to tg.

**Comment by Zergspower** [comment]
I'll be real chief, it's pointless to buy the modsuit for anything but the ash module. The first thing you do is empty everything out except for storage/gps/ash - and given the cons of having less armor with normal speed while it's deployed there's zero reason to even get it. Sphere's a gimmick that is nichely used

**Comment by Hatterhat** [comment]
> it's pointless to buy the modsuit for anything but the ash module. The first thing you do is empty everything out except for storage/gps/ash - and given the cons of having less armor with normal speed while it's deployed there's zero reason to even get it. Sphere's a gimmick that is nichely used and lol of course it's not space proof

MOD ore satchel frees an inventory space since you don't need an ore satchel of holding. Sphere lets you cross lava without having to bring an RCD/lava rods/whatever. Having all that extra complexity means you can install whatever.

If you want to travel light, then travel light. If you want the additional utility of having MOD modules, bring the mining MOD. If you just buy the MOD for the free speed and nothing else, you are welcome to start reconsidering how you use the MOD.

**Comment by Hatterhat** [comment]
Updated the PR body, but for people who've been keeping track with the PR since it started:
- Separated the mining MOD's ash accretion module from the buyable one.
- Buffed the mining MOD's ash accretion module - now it cancels out the suit's slowdown at full ash (meaning you move at full speed, like if you weren't wearing a MOD at all), and provides enough melee armor to be on-par with the explorer suit.
- **The buyable ash accretion module does not receive the above.**
- Did some code cleanup, moving some overrides to the master_files of modular_nova.

**Comment by Zergspower** [comment]
That's a lot better, though losing the speed increase isnt as fortunate. 

Why would the imported one not have the same effect?

**Comment by chemistrymain2** [comment]
> That's a lot better, though losing the speed increase isnt as fortunate.
> 
> Why would the imported one not have the same effect?

It's a bug and should be fixed.

**Comment by TripleShades** [comment]
> I'll be real chief, it's pointless to buy the modsuit for anything but the ash module. The first thing you do is empty everything out except for storage/gps/ash - and given the cons of having less armor with normal speed while it's deployed there's zero reason to even get it. Sphere's a gimmick that is nichely used and lol of course it's not space proof

Nearly every miner I've seen clear their modsuit out for is a power gamer. The point of joining up as a miner is to do your job, which is mining, and the mods that come stock with the mining modsuit are based around this. You do not require extra speed beyond what it gives you to do this, and you do not even require that to mine.

I, personally, get the mining mod because it means I can put various other things in my pocket or belt that cant normally fit there because other items are taking those slots (gps, mining bag) as well as for the suit slot which can carry a Crusher, something I think almost nothing else allows including the Ash Drake set of armor. As for the sphere module, it can absolutely be used for combat, since it comes with the ability to launch small bombs, which do AOE damage, even though it is weak the range is amazing for clearing safely.

Your way is not the only way to play, and this is a longstanding bug that has become normalized.

**Comment by Hatterhat** [comment]
> That's a lot better, though losing the speed increase isnt as fortunate. 

The purpose of this PR is to kill the speed increase, because it's a bug. It is unintended behavior that only *happened* because of retractable MODsuits. If you want the benefit of the MODsuit - the *utility modules*, you use it. If you want better armor, wear better armor. If you want speed boosts, do drugs.

> Why would the imported one not have the same effect? 

Because I envisioned some jackass running a sec MOD on Lavaland for the armorstacking and speed boost and wanted to tell him no ahead of time.

**Comment by Namwty** [comment]
Point is, just cause it was unintended first doesn't mean it has to be fixed. Imo it's a good feature, it makes miners spend less time down at lavaland and more time on station chilling, same as Arcmining does.

Modsuit exactly ain't good and has it's flaws ( most of it is clunkyness, like odd interactions with checking ore amount in the storage module and oxygen tank being moved to bag, but still needing to reopen it to be seen and so on ), I wouldn't exactly call suit that eats plasma to charge exactly benefitial for mining, you literally lose ore by using it.
Yeah sphere is fun, but we literally don't have ruins like pirate cove or russian base, but at best it's niche and that's it.

Like only actual use is bubblegum, everywhere else... it's QoL, to finish mining and be on station chilling.

Zerg is bringing decent point, most people buy the modsuit for the speed, heck I've seen people remove stuff like ore storage cause it's clunky compared to satchel.
It's reasonable to bring this up, mining already has lot's of items that aren't used, no reason to move something closer to "tracking implant" level.

> Nearly every miner I've seen clear their modsuit out for is a power gamer.

Yeah, no. Using something for QoL isn't exactly powergaming.

Miming has plenty of stuff like regen cores, memento mori and cursed hearts, that would be closer to powergaming level. Some of it needs points to obtain, which you will lose by buying mod in first place.

Nor do security have fights on lavaland against someone with that modsuit.

> The point of joining up as a miner is to do your job, which is mining

And speed makes it faster to do and be done with tbf.

And to everyone generally, mining isn't just mining, otherwise we wouldn't have fauna, ruins or megafaunas. Cargo wouldnt have option to sell relics and what not.
Generally could go and go about it.

Frankly it's mindmelting to hear people say "Uhhh miners should only mine" from people be it here or on discord, whenever miner decides to spend 2 minutes killing a drake or a tendrill or even better talking to a ghost role like ashwalker for 5 minutes.


TL;DR
Mining mod is meh, making it more meh is questionable. Even if it is a bug, just leave it as a feature.
It makes miners be able to chill faster on station, due to ore collection speed with movement. Mining as 4 for station of 90 players of whom 5 want to build new bars using ORM is something else.

**Comment by Hatterhat** [comment]
> TL;DR Mining mod is meh, making it more meh is questionable.
The PR basically just makes it a better explorer suit in the intended circumstances (doing your job as a miner). You get MOD functionality. You don't get the slowdown associated with other MODs as long as you have full ash accretion. You get some free inventory slots.

> Even if it is a bug, just leave it as a feature.
It's an exploit that nobody's ever really figured out how to patch until now, so it goes in the bin.

> It makes miners be able to chill faster on station, due to ore collection speed with movement.
If a miner wants to chill on station, they can go off-duty. If they want to work, they can work with tools that don't have exploits.

> Mining as 4 for station of 90 players of whom 5 want to build new bars using ORM is something else.
I think there should only be three miner slots, but that's a config change.

**Comment by Zergspower** [comment]
> > That's a lot better, though losing the speed increase isnt as fortunate.
> 
> The purpose of this PR is to kill the speed increase, because it's a bug. It is unintended behavior that only _happened_ because of retractable MODsuits. If you want the benefit of the MODsuit - the _utility modules_, you use it. If you want better armor, wear better armor. If you want speed boosts, do drugs.

Not everyone can use drugs, it's ironically the 1 item synths get a speedboost from in the sea of stimulants available to everyone. but frankly with that logic all modules shouldn't work when the suit is retracted.

> > Why would the imported one not have the same effect?
> 
> Because I envisioned some jackass running a sec MOD with no slowdown on Lavaland for the armorstacking and I wanted to tell him no ahead of time.

That's a fairly niche situation though, and not something that should be put forth as reasoning for its removal?

This PR seems like a weirdly targetted one with the other various modules modsuits have that still function retracted, if you've used the suit and mine frequently the pattern of its use is clearly there and yes a lot of others myself included even get rid of the terribly made ore module.

btw we've always have had 4 mining slots

**Comment by Zergspower** [comment]
As an aside, for all curious, ARC Mining was balanced for TG and we're working on addressing it for larger codebases. Their balance was done using their data for their servers usage and as such the PR wont work as-is for Nova

**Comment by Namwty** [comment]
> The PR basically just makes it a better explorer suit in the intended circumstances (doing your job as a miner). You get MOD functionality. You don't get the slowdown associated with other MODs as long as you have full ash accretion. You get some free inventory slots.
 
I'm not saying that mining mod is bad and unusable per say, but codebase had hardsuits basically forcefully replaced with modsuits with main feature of it being unique and what not. But all we seeing is, invisible gps/clunky ore storage and resprited explorer suit with storm prot. That slows you down on station. At least before it would be "Low armor, but mobile" which even then... it's only ok in that regard.

Could remove SEVA, which would make it more unique and make shelters usefull, but that is also meh choice, cause slows down mining and other stuff.

> It's an exploit that nobody's ever really figured out how to patch until now, so it goes in the bin

Doesn't exactly explain why it should go, at this rate it could be just a counter to "codebase movement speed diff" for miners at very least.
Same as we tweaked dswords for bitrunners, cause codebase diff.
Like when I see a PR and just see "Reminds miners that they are mortal" as half of "How this is a good change" isn't exaclty giving hope, more gives avarage "Fuck miners" vibes than anything.

Yes is it a bug fix? Sure. Is it a good bug to fix? Arguable.

This bug fix just makes mining slower, which effects lots of people on station, that depend on mats existing in ORM and ideally at decent pace.

Same with pre nerf "Fuck miners" vileworms, people realized that sitting half of the round without mats, cause miners kept dying, wasn't that fun actually.

> If a miner wants to chill on station, they can go off-duty.

Some people want to just chill after 1 hour mark, but still want to help incase mats vanish to hell to like vortex and what not or some newbie miner got chewed by ash drake. Like If some miner dies to something, I'll go and recover them. If someone suddenly needs clay/lazarus/set up BSA target gps etc, I will help. Who are people going to ask if every miner went off duty.

> I think there should only be three miner slots, but that's a config change.

Tg balanced mining poorly ( see ice box and lavaland ore ratio for example ), thankfully arcminin will be a step in some direction... good or bad dunno yet ( bad without change for us due 3 hour rounds )
You can't compare Nova stations mat consumption and Manuels/Terry/Yog/Bee, it's all different. We have more people building and renovating stuff than Terry as an example.
Then there are factors like new player and someone training said new player or bubble spawning 10 meters away from outpost and chewing miner instantly.

Terry couldn't care less if 3 miners died to random crap, cause round time. We do care, cause no one on station wants to sit 3 hours without mats, same like no one wants to sit on station when sm delams within 5 mins. The more miners the less chances of some miner being soft RR for 2 hours cause no one can find them.

Regardless, lowering mining slots would be bad for basically everyone.
I could talk about para220 having 9 slots or whatever, but they do have different factors like codebase and mat consumption diff.


Regardless I never use mining mod myself regardless with or without this change, but kinda ouch seeing it be explorer suit+ considering some folk do use it, some would maybe stop.
If it gets changed it's whatever, mining has enough gear variety as it is.

**Comment by Hatterhat** [comment]
> Not everyone can use drugs, it's ironically the 1 item synths get a speedboost from in the sea of stimulants available to everyone. but frankly with that logic all modules shouldn't work when the suit is retracted.

The most grievous offender here is the speedboost exploit, and the PR fixes the exploit. The rest are inconsequential enough where I don't really care about it. If you buy the mining MOD, you should probably be using the mining MOD as intended (a full suit) and not as a retracted control module on your back.

> This PR seems like a weirdly targetted one with the other various modules modsuits have that still function retracted, if you've used the suit and mine frequently the pattern of its use is clearly there and yes a lot of others myself included even get rid of the terribly made ore module.

Good news! I'm not encouraging you to gut your MOD, that's entirely on you. I'm encouraging you to actually keep it deployed, and switch it out if/when you get better armor. I don't care what you do with the rest of the modules, just put the rest of the suit on. It's nonsense that it gives you a speedboost while retracted, and the reason it only did that is because of unintended behavior that this PR is addressing.

> btw we've always have had 4 mining slots

Maps have three lockers, not counting the one in auxbase, which is there for greytiders to ransack or whatever. Regardless, it's a config change to reduce mining slots, but I'd personally like to see it happen anyway.

--- 
> Who are people going to ask if every miner went off duty.

They can go to the public mining shuttle themselves.

> Regardless I never use mining mod myself regardless with or without this change, but kinda ouch seeing it be explorer suit+ considering some folk do use it, some would maybe stop.

If people want to stop using their extra job gear that they purchased, more power to them. They can spend their mining points on credit stacks or something. If you don't use the MOD, why do you care about it?

**Comment by W4rd3nn** [comment]
Consider most if not all modules are unusable if the suit doesn't have at least the chestplate deployed, I'd say a speedboost applied when no pieces are deployed is a bug.

**Comment by Ephemeralis** [comment]
It is *very clearly* a bug. It is insane to me that people are writing small essays in here defending the fix of obviously unintended functionality.

**Comment by AdamElTablawy** [comment]
ITT powergamers getting mad their bug is being fixed because they can’t abuse it any more

File: modular_nova/master_files/code/modules/mod/modules/modules_supply.dm [hunk 1]
```diff
@@ -5,3 +5,74 @@
...
+/obj/item/mod/module/ash_accretion/on_suit_deactivation(deleting)
+	. = ..()
+	protection_enabled = FALSE
+	UnregisterSignal(mod, list(COMSIG_MOD_DEPLOYED, COMSIG_MOD_RETRACTED))
+
+/obj/item/mod/module/ash_accretion/proc/update_added_speed()
+	SIGNAL_HANDLER
+	speed_added = mod.slowdown_active // no, you don't get to have a free speedup, actually
+
+/obj/item/mod/module/ash_accretion/proc/on_mod_toggle()
```
Comment (GoldenAlpharex) [inline modular_nova/master_files/code/modules/mod/modules/modules_supply.dm:41]: Document these procs, please.

File: modular_nova/master_files/code/modules/mod/modules/modules_supply.dm [hunk 2]
```diff
@@ -5,3 +5,74 @@
...
+	if(!traveled_tiles)
+		return
+	var/list/parts = mod.mod_parts + mod
+	var/datum/armor/to_remove = get_armor_by_type(armor_mod)
+	for(var/obj/item/part as anything in parts)
+		part.set_armor(part.get_armor().subtract_other_armor(to_remove.generate_new_with_multipliers(list(ARMOR_ALL = traveled_tiles))))
+	if(traveled_tiles == max_traveled_tiles)
+		mod.slowdown += speed_added
+		mod.wearer.update_equipment_speed_mods()
+	traveled_tiles = 0
```
Comment (GoldenAlpharex) [inline modular_nova/master_files/code/modules/mod/modules/modules_supply.dm:67]: This isn't a blocking comment or anything, but moving forward, try your hand at making your code breathe a bit better. This just feels like one big block of code and makes it a lot harder to review and read through.

Here's a suggestion of how to spread it out a bit better, to improve the readability.
```suggestion
	if((mod.wearer.head == mod.helmet) && (mod.wearer.wear_suit == mod.chestplate) && (mod.wearer.gloves == mod.gauntlets) && (mod.wearer.shoes == mod.boots) && mod.active)
		// suit is on and fully deployed, give them their proofing
		mod.wearer.add_traits(list(TRAIT_ASHSTORM_IMMUNE, TRAIT_SNOWSTORM_IMMUNE), MOD_TRAIT)
		RegisterSignal(mod.wearer, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
		balloon_alert(mod.wearer, "ash accretion enabled")
		protection_enabled = TRUE
		return

	// if their suit is not fully deployed, take their proofing away
	if(!protection_enabled)
		return // unless it was already gone

	UnregisterSignal(mod.wearer, COMSIG_MOVABLE_MOVED)
	mod.wearer.remove_traits(list(TRAIT_ASHSTORM_IMMUNE, TRAIT_SNOWSTORM_IMMUNE), MOD_TRAIT)
	balloon_alert(mod.wearer, "ash accretion disabled!")
	protection_enabled = FALSE
	if(!traveled_tiles)
		return

	var/list/parts = mod.mod_parts + mod
	var/datum/armor/to_remove = get_armor_by_type(armor_mod)
	for(var/obj/item/part as anything in parts)
		part.set_armor(part.get_armor().subtract_other_armor(to_remove.generate_new_with_multipliers(list(ARMOR_ALL = traveled_tiles))))

	if(traveled_tiles == max_traveled_tiles)
		mod.slowdown += speed_added
		mod.wearer.update_equipment_speed_mods()

	traveled_tiles = 0
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I'm totally for this PR, it's funny to see people complaining that we're finally fixing a bug that's overstayed its welcome, and that people shouldn't've been relying on to begin with, because it's always been clearly unintended.

File: modular_nova/master_files/code/modules/mod/modules/modules_supply.dm [hunk 3]
```diff
@@ -5,3 +5,81 @@
...
+	var/datum/armor/to_remove = get_armor_by_type(armor_mod)
+	for(var/obj/item/part as anything in parts)
+		part.set_armor(part.get_armor().subtract_other_armor(to_remove.generate_new_with_multipliers(list(ARMOR_ALL = traveled_tiles))))
+
+	if(traveled_tiles == max_traveled_tiles)
+		mod.slowdown += speed_added
+		mod.wearer.update_equipment_speed_mods()
+
+	traveled_tiles = 0
+		mod.wearer.update_equipment_speed_mods()
```
Comment (GoldenAlpharex) [inline modular_nova/master_files/code/modules/mod/modules/modules_supply.dm:73]:
```suggestion
```

**Comment by Hatterhat** [comment]
> OK, having glanced this over, it looks like the speed concerns have been addressed.
> 
> However I don't understand why this needs to be the case:
> 
> image
> 
> Can probably just get rid of that part and have them all function the same to avoid confusion.

image

If the import module had the same buffs as the mining MOD exclusive one, you could run a sec MOD and get 75 melee armor with no slowdown, along with the whole host of buffs that it grants (stormproofing, other armor, etc. etc). At this point in time, though, the external ash accretion module's speedboost actually... *does* cancel out the security MOD's 0.5 active slowdown. I don't think I'm actually too fond of that concept. I'll think about it some more.

**Comment by Hatterhat** [comment]
I have exploded the import version of the module.
